### PR TITLE
Add id and comment

### LINKED ISSUE #10: Add Sequence.id attribute
and also `Sequence.comment` and perhaps `Sequence.sep`. They could be lazily computed. Perhaps also add `.header` as an alias for `.name`? Should setting `.id` and `.comment` be supported? Convenient, but perhaps slower than setting the full header directly.

### PR DESCRIPTION
**Pull request by rhpvorderman**
You had something like this in mind?

EDIT:
fixes #10

### CODE REVIEW

File: src/dnaio/_core.pyx [hunk 1]
```diff
@@ -150,6 +154,47 @@ cdef class SequenceRecord:
...
+        # Not yet cached is None
+        if self._comment is None:
+            name = <char *>PyUnicode_DATA(self._name)
+            name_length = <size_t>PyUnicode_GET_LENGTH(self._name)
+            id_length = strcspn(name, "\t ")
+            if id_length == name_length:
+                self._comment = ""
+            else:
+                comment_length = name_length - (id_length + 1)
+                comment_start = name + id_length + 1
```
Comment (marcelm) [inline src/dnaio/_core.pyx:190]: Hm, this assumes that there’s only one whitespace between the id and the comment. I’d assume this is true most of the time, but I think we need to be a bit more defensive.

File: src/dnaio/_core.pyx [hunk 2]
```diff
@@ -150,6 +154,47 @@ cdef class SequenceRecord:
...
+                self._comment = PyUnicode_New(comment_length , 127)
+                memcpy(PyUnicode_DATA(self._comment), comment_start, comment_length)
+        # Cached but nothing is internally empty string, expose externally as None
```
Comment (marcelm) [inline src/dnaio/_core.pyx:193]: I don’t get the comment. Maybe just remove? Would be ok

File: tests/test_records.py [hunk 3]
```diff
@@ -105,6 +105,50 @@ def test_set_qualities_none(self):
...
+            seq.comment = "Hello there!"
+
+    @pytest.mark.parametrize(
+        ["record", "expected"],
+        [
+            (SequenceRecord("name", "A", "="), None),
+            (SequenceRecord("name ", "A", "="), None),
+            (SequenceRecord("name  ", "A", "="), " "),
+            (SequenceRecord("name", "A", "="), None),
+            (SequenceRecord("AotC I hate sand!", "A", "="), "I hate sand!"),
```
Comment (marcelm) [inline tests/test_records.py:125]: :smile:

**Review by marcelm — COMMENTED** [review]
Yes, very nice! This will be really nice to have. We should make a release as soon as this is merged.

This also needs a changelog and documentation update. (I can do that if you don’t have time over the coming days.)

File: src/dnaio/_core.pyx (hunk 1 above)
Comment (rhpvorderman) [reply src/dnaio/_core.pyx:190]: Finally, a use for strspn!

File: src/dnaio/_core.pyx (hunk 2 above)
Comment (rhpvorderman) [reply src/dnaio/_core.pyx:193]: I will try to make it more clear in the comment. It is a rather odd design choice so it needs one, I think.

File: tests/test_records.py (hunk 3 above)
Comment (rhpvorderman) [reply tests/test_records.py:125]: https://www.youtube.com/watch?v=C_yWiW7OxZw

**Comment by rhpvorderman** [comment]
All done!

File: tests/test_records.py (hunk 3 above)
Comment (marcelm) [reply tests/test_records.py:125]: Thanks, I got the Star Wars reference, but didn’t know the song (and know it’ll be stuck in my head for the rest of the day ...)

Comment (rhpvorderman) [reply tests/test_records.py:125]: Well you could always listen to "Last jedi knight" by the same musician to get it out of your head again ;-).

**Pull request merged**